File: src/org/tec/datastructures/linear/SimpleList.java
package org.tec.datastructures.linear;

import org.tec.datastructures.nodes.NodoSimple;


public class SimpleList<T extends Comparable<T>> {
	private NodoSimple<T> primero;
	int numeroElementos;
	
	public SimpleList() {
		this.primero = null;
		this.numeroElementos = 0;
	}
	
	public boolean isEmpty() {
		return this.primero == null;
	}

	public NodoSimple<T> getFirst() {
		return primero;
	}

	public void setFirst(NodoSimple<T> primero) {
		this.primero = primero;
	}
	
	public void printList() {
		NodoSimple<T> actual = this.primero;
		while ( actual != null) {
			System.out.println(actual.getValue());
			actual = actual.getNext();
		}
	}
	
	public void insertFirst (T valor) {
		if (this.primero == null) {
			this.primero = new NodoSimple<T>(valor);
			this.numeroElementos++;
		} else {
			NodoSimple<T> nuevo = new NodoSimple<T>(valor);
			nuevo.setNext(this.primero);
			this.primero = nuevo;
			this.numeroElementos++;
		}
		
	}
	
	public void insertEnd (T valor) {
		if (this.primero == null) {
			this.primero = new NodoSimple<T>(valor);
			this.numeroElementos++;
		} else {
			NodoSimple<T> actual = this.primero;
			while( actual.getNext() != null) {
				actual = actual.getNext();
			}
			actual.setNext(new NodoSimple<T>(valor));
		}	this.numeroElementos++;
		
	}
	
	public NodoSimple<T> search(T buscado) {
		NodoSimple<T> actual = this.primero;
		while(actual != null) {
			if (actual.getValue().compareTo(buscado) == 0) {
				return actual;
			} else {
				actual = actual.getNext();
			}
		}
		return null;
	}
	
	public void remove (T valor) {
		if (search(valor) != null) {
			if (this.primero.getValue() == valor) {
				this.primero = this.primero.getNext();
				this.numeroElementos--;
			} else {
				if (this.primero != null) {
					NodoSimple<T> anterior = this.primero;
					while( anterior.getNext().getValue() != valor){
						anterior = anterior.getNext();
					}
					NodoSimple<T> actual = anterior.getNext().getNext();
					anterior.setNext(actual);  
				}
        	 
			}
			this.numeroElementos--;
		
		}
		
	}
	
	public int getSize () {
		return this.numeroElementos;
	}
	
	
}



		

File: Rider/org.tec.datastructures/Linear/SimpleList.cs

﻿using System;
using org.tec.datastructures.Nodes;

namespace org.tec.datastructures.Linear
{
    public class SimpleList<T> where T : IComparable<T>{
        
        private SimpleNode<T> _head;
	
        public SimpleList() {
            _head = null;
        }
	
        public SimpleList(SimpleNode<T> head) {
            _head = head;
        }
	
        public SimpleNode<T> GetHead(){
            return _head;
        }
	
	
        public void Append (T value) {
            if (_head == null){
                _head = new SimpleNode<T>(value);
                return;
            }
            SimpleNode<T> temp = _head;
            while (temp.HasNext()){
                temp = temp.GetNext();
            }
            temp.SetNext(new SimpleNode<T>(value));
        }

	
        public SimpleNode<T> Search(T value) {
            SimpleNode<T> temp = _head;
            while(temp != null){
                if(temp.GetValue().Equals(value)){
                    break;
                }
                temp = temp.GetNext();
            }
            return temp;
        }

        public void Delete(T value) {
            SimpleNode<T> temp = _head;
            SimpleNode<T> prev = null;
            while(temp != null){
                if(temp.GetValue().Equals(value)){
                    if (temp == _head){
                        _head = temp.GetNext();
                    }
                    else{
                        prev.SetNext(temp.GetNext());
                    }
                    break;
                }
                prev = temp;
                temp = temp.GetNext();
            }
		
        }		
	
	
        public bool IsEmpty(){
            return _head == null;
        }
	
	
        public int Length() {
            SimpleNode<T> current = _head;
            int result = 0;
            while (current != null){
                result++;
                current = current.GetNext();
            }
            return result;
        }

        public void Clear() {
            _head = null;
        }
    }

  
}